### PR TITLE
[SPARK-41478][SQL] Assign a name to the error class _LEGACY_ERROR_TEMP_1234

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
In the PR, I propose to assign the name ANALYZE_UNCACHED_TEMP_VIEW to the error class _LEGACY_ERROR_TEMP_1234.

### Why are the changes needed?
Proper names of error classes should improve user experience with Spark SQL.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Add new UT.
Pass GA.

### CODE REVIEW

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1257,6 +1257,11 @@
           "AES-<mode> with the padding <padding> by the <functionName> function."
         ]
       },
+      "ANALYZE_TABLE_VIEW_NOT_CACHED" : {
+        "message" : [
+          "Temporary view <tableName> is not cached for analyzing columns."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:1262]: Could you improve the error message by following the common structure as in other errors in `error-classes.json`:
1. State the issue
2. Give a suggestion how to solve it.

The error could be:
```suggestion
          "The ANALYZE TABLE FOR COLUMNS command can operate on temporary views that have been cached already. Consider to cache the view <view Name>."
```

File: core/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -1257,6 +1257,11 @@
           "AES-<mode> with the padding <padding> by the <functionName> function."
         ]
       },
+      "ANALYZE_TABLE_VIEW_NOT_CACHED" : {
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:1260]:
```suggestion
      "ANALYZE_UNCACHED_TEMP_VIEW" : {
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: core/src/main/resources/error/error-classes.json (hunk 2 above)
Comment (panbingkun) [reply core/src/main/resources/error/error-classes.json]: Done

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Pull request closed without merging**